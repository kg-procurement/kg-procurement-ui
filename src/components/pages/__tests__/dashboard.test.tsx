import { fireEvent, render, screen } from '@testing-library/react'

import DashboardPage from '../dashboard.tsx'

describe('<DashboardPage />', () => {
  it('should render the header section with the logo', () => {
    render(<DashboardPage />)

    const logo = screen.getByAltText('Kompas Gramedia Logo Background')
    expect(logo).toBeInTheDocument()
    expect(logo).toHaveAttribute('src', 'kompas-gramedia-logo-bg.svg')

    expect(screen.getByText('Vendor A')).toBeInTheDocument()
    expect(screen.getByText('Dashboard')).toBeInTheDocument()
  })

  it('should render the table with correct columns and rows', () => {
    render(<DashboardPage />)

    expect(screen.getByText('Name')).toBeInTheDocument()
    expect(screen.getByText('Description')).toBeInTheDocument()
    expect(screen.getByText('Price')).toBeInTheDocument()

    const rows = screen.getAllByRole('row')
    expect(rows).toHaveLength(6)

    expect(screen.getByText('Lorem Ipsum Amer')).toBeInTheDocument()
    expect(screen.getByText('Rp.25000')).toBeInTheDocument()
  })

  it('should render the footer', () => {
    render(<DashboardPage />)
    expect(screen.getByText(/Â© 2024 KOMPAS/i)).toBeInTheDocument()
  })

  it('should render icons in the header', () => {
    render(<DashboardPage />)

    const icons = screen.getAllByRole('img')
    expect(icons).toHaveLength(2)
  })

  it('should render the edit icon in the header', () => {
    render(<DashboardPage />)

    const editIconButton = screen.getByRole('button', { hidden: true })
    expect(editIconButton).toBeInTheDocument()
  })

  it('should open the dialog', () => {
    const { queryByRole } = render(<DashboardPage />)

    const editIconButton = screen.getByRole('button', { hidden: true })
    fireEvent.click(editIconButton)
    expect(queryByRole('dialog')).toBeInTheDocument()
  })

  it('should open and close the dialog', () => {
    const { getByText, queryByRole } = render(<DashboardPage />)

    const editIconButton = screen.getByRole('button', { hidden: true })
    fireEvent.click(editIconButton)
    expect(queryByRole('dialog')).toBeInTheDocument()

    fireEvent.click(getByText('Cancel'))
    expect(queryByRole('dialog')).not.toBeInTheDocument()
  })
})
