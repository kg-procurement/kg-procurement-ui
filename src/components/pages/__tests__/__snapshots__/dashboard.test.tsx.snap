// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<DashboardPage /> > should render the table with correct columns and rows 2`] = `
"Vendor Aundefined
Dashboard
undefined
undefined
undefined
Inventory List
NameDescriptionPriceMP3 & MP4 PlyerMP3 & MP4 PlyerRp.25000undefinedOvenOvenRp.25000undefinedSling BagSling BagRp.25000undefinedundefinedPrevious1undefinedMore pagesNext
undefinedÂ© 2024 KOMPAS
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
Contact us (Email):"
`;
